<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Creates an object of class mvnmm. — fit • lineaGT</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Creates an object of class mvnmm. — fit"><meta name="description" content="Function to fit the input data."><meta property="og:description" content="Function to fit the input data."><meta property="og:image" content="caravagnalab.github.io/lineaGT/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">lineaGT</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/lineaGT.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Inference.html">Lineage inference</a></li>
    <li><a class="dropdown-item" href="../articles/Input-formats.html">Input formats</a></li>
    <li><a class="dropdown-item" href="../articles/Plotting-functions.html">Plotting functions</a></li>
    <li><a class="dropdown-item" href="../articles/Utilities.html">Utilities</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/lineaGT/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Creates an object of class <code>mvnmm</code>.</h1>
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/lineaGT/blob/main/R/fit_model.R" class="external-link"><code>R/fit_model.R</code></a></small>
      <div class="d-none name"><code>fit.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Function to fit the input data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">fit</span><span class="op">(</span></span>
<span>  <span class="va">cov.df</span>,</span>
<span>  vaf.df <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  infer_phylogenies <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  infer_growth <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  k_interval <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fl">15</span><span class="op">)</span>,</span>
<span>  n_runs <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  steps <span class="op">=</span> <span class="fl">500</span>,</span>
<span>  min_steps <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  lr <span class="op">=</span> <span class="fl">0.005</span>,</span>
<span>  p <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  min_frac <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  max_IS <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  check_conv <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  covariance <span class="op">=</span> <span class="st">"full"</span>,</span>
<span>  hyperparams <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  default_lm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  timepoints_to_int <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  show_progr <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  store_grads <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  store_losses <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  store_params <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  seed_optim <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  seed_init <span class="op">=</span> <span class="fu">reticulate</span><span class="fu">::</span><span class="fu"><a href="https://rstudio.github.io/reticulate/reference/py_none.html" class="external-link">py_none</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  sample_id <span class="op">=</span> <span class="st">""</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-cov-df">cov.df<a class="anchor" aria-label="anchor" href="#arg-cov-df"></a></dt>
<dd><p>Input coverage dataset. It must have at least the columns <code>coverage</code>, and <code>IS</code>,
and additional columns <code>timepoints</code> and <code>lineage</code>, will be added if missing, assuming single
timepoint and lineage.</p></dd>


<dt id="arg-vaf-df">vaf.df<a class="anchor" aria-label="anchor" href="#arg-vaf-df"></a></dt>
<dd><p>Input VAF dataset. If not <code>NULL</code>, the mutations clustering will be performed.
It must have at least the columns <code>mutation</code>, <code>IS</code>, <code>alt</code>, <code>dp</code>, and additional
<code>vaf</code>, <code>timepoints</code>, <code>lineage</code>, <code>IS</code>, <code>mutation</code>, with the number of reads for the mutated allele,
overall depth, vaf values, timepoint, lineage, IS and mutation, respectively.</p></dd>


<dt id="arg-infer-phylogenies">infer_phylogenies<a class="anchor" aria-label="anchor" href="#arg-infer-phylogenies"></a></dt>
<dd><p>A Boolean. If set to <code>TRUE</code>, the function will also compute and attach to
the returned object the phylogenetic trees for each cluster.</p></dd>


<dt id="arg-k-interval">k_interval<a class="anchor" aria-label="anchor" href="#arg-k-interval"></a></dt>
<dd><p>Interval of K values to test.</p></dd>


<dt id="arg-n-runs">n_runs<a class="anchor" aria-label="anchor" href="#arg-n-runs"></a></dt>
<dd><p>Number of runs to perform for each K.</p></dd>


<dt id="arg-steps">steps<a class="anchor" aria-label="anchor" href="#arg-steps"></a></dt>
<dd><p>Maximum number of steps for the inference.</p></dd>


<dt id="arg-lr">lr<a class="anchor" aria-label="anchor" href="#arg-lr"></a></dt>
<dd><p>Learning rate used in the inference.</p></dd>


<dt id="arg-p">p<a class="anchor" aria-label="anchor" href="#arg-p"></a></dt>
<dd><p>Numeric value used to check the convergence of the parameters.</p></dd>


<dt id="arg-min-frac">min_frac<a class="anchor" aria-label="anchor" href="#arg-min-frac"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-max-is">max_IS<a class="anchor" aria-label="anchor" href="#arg-max-is"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-check-conv">check_conv<a class="anchor" aria-label="anchor" href="#arg-check-conv"></a></dt>
<dd><p>A Boolean. If set to <code>TRUE</code>, the function will check for early convergence,
otherwise it will perform <code>steps</code> iterations.</p></dd>


<dt id="arg-covariance">covariance<a class="anchor" aria-label="anchor" href="#arg-covariance"></a></dt>
<dd><p>Covariance type for the Multivariate Gaussian.</p></dd>


<dt id="arg-hyperparams">hyperparams<a class="anchor" aria-label="anchor" href="#arg-hyperparams"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-default-lm">default_lm<a class="anchor" aria-label="anchor" href="#arg-default-lm"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-timepoints-to-int">timepoints_to_int<a class="anchor" aria-label="anchor" href="#arg-timepoints-to-int"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-show-progr">show_progr<a class="anchor" aria-label="anchor" href="#arg-show-progr"></a></dt>
<dd><p>A Boolean. If <code>TRUE</code>, the progression bar will be shown during inference.</p></dd>


<dt id="arg-store-grads">store_grads<a class="anchor" aria-label="anchor" href="#arg-store-grads"></a></dt>
<dd><p>A Booolean. If <code>TRUE</code>, the gradient norms for the parameters at each
iteration will be stored.</p></dd>


<dt id="arg-store-losses">store_losses<a class="anchor" aria-label="anchor" href="#arg-store-losses"></a></dt>
<dd><p>A Boolean. If <code>TRUE</code>, the computed losses for the parameters at each
iteration will be stored.</p></dd>


<dt id="arg-store-params">store_params<a class="anchor" aria-label="anchor" href="#arg-store-params"></a></dt>
<dd><p>A Boolean. If <code>TRUE</code>, the estimated parameters at each iteration will be stored.</p></dd>


<dt id="arg-seed-optim">seed_optim<a class="anchor" aria-label="anchor" href="#arg-seed-optim"></a></dt>
<dd><p>add</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Value of the seed.</p></dd>


<dt id="arg-sample-id">sample_id<a class="anchor" aria-label="anchor" href="#arg-sample-id"></a></dt>
<dd><p>add</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A <code>mvnmm</code> object, containing the input dataset, annotated with IS_values, N, K, T
specific of the dataset, the input IS and column names, a list params that will contain the
inferred parameters, the python object</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elena Buscaroli, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

