<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Plotting functions • lineaGT</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plotting functions">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">lineaGT</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/lineaGT.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/Inference.html">Lineage inference</a></li>
    <li><a class="dropdown-item" href="../articles/Input-formats.html">Input formats</a></li>
    <li><a class="dropdown-item" href="../articles/Plotting-functions.html">Plotting functions</a></li>
    <li><a class="dropdown-item" href="../articles/Utilities.html">Utilities</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/caravagnalab/lineaGT/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Plotting functions</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagnalab/lineaGT/blob/main/vignettes/articles/Plotting-functions.Rmd" class="external-link"><code>vignettes/articles/Plotting-functions.Rmd</code></a></small>
      <div class="d-none name"><code>Plotting-functions.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/caravagnalab/lineaGT" class="external-link">lineaGT</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'VIBER'</span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'cli::num_ansi_colors' by</span></span>
<span><span class="co">#&gt; 'crayon::num_ansi_colors' when loading 'easypar'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">ctree</span>, <span style="font-style: italic;">'Clone trees in cancer'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/ctree/&gt;</span></span></span>
<span><span class="co">#&gt; Warning: replacing previous import 'crayon::%+%' by 'ggplot2::%+%' when loading</span></span>
<span><span class="co">#&gt; 'VIBER'</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">VIBER</span>, <span style="font-style: italic;">'Variational inference for multivariate Binomial mixtures'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagn.github.io/VIBER/&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Loading <span style="color: #00BB00;">lineaGT</span>, <span style="font-style: italic;">'Lineage inference from gene therapy'</span>. Support : <span style="color: #0000BB; font-style: italic;">&lt;https://caravagnalab.github.io/lineaGT/&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> The 'lineagt-env' environment is already loaded!</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span>
<span><span class="va">x.example</span></span>
<span><span class="co">#&gt; ── <span style="color: #000000; background-color: #BBBB00;">[ lineaGT ] </span> ──── <span style="font-weight: bold;">Python:</span> <span style="color: #555555;">/usr/share/miniconda/envs/lineagt-env/bin/python</span> ──</span></span>
<span><span class="co">#&gt; → Lineages: <span style="color: #00BB00;">l1</span> and <span style="color: #00BB00;">l2</span>.</span></span>
<span><span class="co">#&gt; → Timepoints: <span style="color: #00BB00;">t1</span> and <span style="color: #00BB00;">t2</span>.</span></span>
<span><span class="co">#&gt; → Number of Insertion Sites: <span style="color: #00BB00;">66</span>.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── Optimal IS model with <span style="color: #00BB00;">k = 8</span>.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     <span style="color: #BBBB00;">C4</span> (19 ISs) : <span style="color: #0000BB;">l1</span> [285, 209]; <span style="color: #0000BB;">l2</span> [ 51, 492] </span></span>
<span><span class="co">#&gt;     <span style="color: #BBBB00;">C1</span> (15 ISs) : <span style="color: #0000BB;">l1</span> [245, 177]; <span style="color: #0000BB;">l2</span> [ 23, 289] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C0</span> (6 ISs) : <span style="color: #0000BB;">l1</span> [145, 240]; <span style="color: #0000BB;">l2</span> [ 32, 373] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C2</span> (6 ISs) : <span style="color: #0000BB;">l1</span> [  1, 547]; <span style="color: #0000BB;">l2</span> [  1, 388] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C3</span> (6 ISs) : <span style="color: #0000BB;">l1</span> [ 92, 109]; <span style="color: #0000BB;">l2</span> [245, 751] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C5</span> (6 ISs) : <span style="color: #0000BB;">l1</span> [  0, 551]; <span style="color: #0000BB;">l2</span> [  1, 828] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C6</span> (4 ISs) : <span style="color: #0000BB;">l1</span> [330,  16]; <span style="color: #0000BB;">l2</span> [ 17,  38] </span></span>
<span><span class="co">#&gt;      <span style="color: #BBBB00;">C7</span> (4 ISs) : <span style="color: #0000BB;">l1</span> [  0, 426]; <span style="color: #0000BB;">l2</span> [  1, 198]</span></span></code></pre></div>
<div class="section level2">
<h2 id="mixture-weights">Mixture weights<a class="anchor" aria-label="anchor" href="#mixture-weights"></a>
</h2>
<p>The mixture weights and number of ISs per cluster can be visualized
with the function <code><a href="../reference/plot_mixture_weights.html">plot_mixture_weights()</a></code> .</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_mixture_weights.html">plot_mixture_weights</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="scatterplot">Scatterplot<a class="anchor" aria-label="anchor" href="#scatterplot"></a>
</h2>
<p>The function <code><a href="../reference/plot_scatter_density.html">plot_scatter_density()</a></code> returns a list of 2D
multivariate densities estimated by the model. The argument
<code>highlight</code> can be used to show only a subset of clusters and
the argument <code>min_frac</code> to show the clusters with the
specified frequency in at least one dimension.</p>
<p>Note that the observed coverage values across lineages and over time
are modeled as independent, therefore each dimension corresponds to a
combination of time-point and lineage.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">plots</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_scatter_density.html">plot_scatter_density</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span>
<span><span class="va">plots</span><span class="op">$</span><span class="va">`cov.t2.l1:cov.t1.l2`</span> <span class="co"># to visualize a single plot</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="marginal-distributions">Marginal distributions<a class="anchor" aria-label="anchor" href="#marginal-distributions"></a>
</h2>
<p>The function <code><a href="../reference/plot_marginal.html">plot_marginal()</a></code> returns a plot with the
marginal estimated densities for each cluster, time-point and
lineage.</p>
<p>The option <code>single_plot</code> returns the density of the whole
mixture grouped by lineage and time-point.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">marginals</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_marginal.html">plot_marginal</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span>
<span><span class="va">marginals_mixture</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_marginal.html">plot_marginal</a></span><span class="op">(</span><span class="va">x.example</span>, single_plot<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">marginals</span> <span class="op">/</span> <span class="va">marginals_mixture</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="mullerplot">Mullerplot<a class="anchor" aria-label="anchor" href="#mullerplot"></a>
</h2>
<p>The function <code><a href="../reference/plot_mullerplot.html">plot_mullerplot()</a></code> shows the expansion of
the identified populations over time. It supports the options
<code>which=c("frac","pop")</code> corresponding to the absolule
population abundance and the relative fraction, respectively.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mp1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, which<span class="op">=</span><span class="st">"frac"</span><span class="op">)</span></span>
<span><span class="va">mp2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, which<span class="op">=</span><span class="st">"pop"</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">mp1</span>, <span class="va">mp2</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>If the option <code>mutations</code> is set to <code>TRUE</code>,
then the subclones originated within each population will be reported as
well in the mullerplot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mp1</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, which<span class="op">=</span><span class="st">"frac"</span>, mutations<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="va">mp2</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, which<span class="op">=</span><span class="st">"pop"</span>, mutations<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">mp1</span>, <span class="va">mp2</span>, ncol<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>The function supports also the visualization of a single clone to
monitor the growth of subpopulations, through the argument
<code>single_clone</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, highlight<span class="op">=</span><span class="st">"C4"</span>, mutations<span class="op">=</span><span class="cn">T</span>, single_clone<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Moreover, some of the identified clusters (showing low coverage in
all dimensions) represents poly-clonal populations, since they cannot be
uniquely identified by the mixture model. Therefore, the estimated
abundance values might be readjusted according to the estimated number
of populations in each clusters.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/estimate_n_pops.html">estimate_n_pops</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span>
<span><span class="co">#&gt; C0 C1 C2 C3 C4 C5 C6 C7 </span></span>
<span><span class="co">#&gt;  1  2  1  1  2  1  1  1</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_mullerplot.html">plot_mullerplot</a></span><span class="op">(</span><span class="va">x.example</span>, which<span class="op">=</span><span class="st">"frac"</span>, mutations<span class="op">=</span><span class="cn">T</span>, estimate_npops<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="vaf">VAF<a class="anchor" aria-label="anchor" href="#vaf"></a>
</h2>
<p>The function <code><a href="../reference/plot_vaf_time.html">plot_vaf_time()</a></code> can be used to visualize
the behaviour of mutations variant allele frequencies over time for each
subclone.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_vaf_time.html">plot_vaf_time</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="phylogenetic-evolution">Phylogenetic evolution<a class="anchor" aria-label="anchor" href="#phylogenetic-evolution"></a>
</h2>
<p>For each cluster of ISs, the function <code><a href="../reference/plot_phylogeny.html">plot_phylogeny()</a></code>
reports the estimated phylogenetic tree.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_phylogeny.html">plot_phylogeny</a></span><span class="op">(</span><span class="va">x.example</span><span class="op">)</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; This graph was created by an old(er) igraph version.</span></span>
<span><span class="co">#&gt;   Call upgrade_graph() on it to use with the current igraph version</span></span>
<span><span class="co">#&gt;   For now we convert it on the fly...</span></span>
<span><span class="co">#&gt; $C0</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $C1</span></span></code></pre>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-12-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $C4</span></span></code></pre>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-12-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $C7</span></span></code></pre>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-12-4.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="clonal-growth">Clonal Growth<a class="anchor" aria-label="anchor" href="#clonal-growth"></a>
</h2>
<p>The fitted exponential and logistic growth regressions are shown with
the <code><a href="../reference/plot_growth_regression.html">plot_growth_regression()</a></code> , reporting by default the fit
of the best model, selected as the one with the highest likelihood.</p>
<p>Both regressions can be inspected setting <code>show_best=F</code>
.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_growth_regression.html">plot_growth_regression</a></span><span class="op">(</span><span class="va">x.example</span>, show_best<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>The same function can be used to show the growth regressions for the
subclones identified by somatic mutations.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_growth_regression.html">plot_growth_regression</a></span><span class="op">(</span><span class="va">x.example</span>, highlight<span class="op">=</span><span class="st">"C4"</span>, mutations<span class="op">=</span><span class="cn">T</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>An alternative way of visualising differences in growth rates is
through <code><a href="../reference/plot_growth_rates.html">plot_growth_rates()</a></code> function, reporting the values
of estimated growth rates for each (sub)population.</p>
<p>Disabling the <code>show_best</code> option, the model with lowest
likelihood is shown as a dashed line.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_growth_rates.html">plot_growth_rates</a></span><span class="op">(</span><span class="va">x.example</span>, show_best<span class="op">=</span><span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="Plotting-functions_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elena Buscaroli, Giulio Caravagna.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
